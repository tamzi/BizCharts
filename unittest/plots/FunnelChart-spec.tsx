import React from 'react';
import FunnelChart from '../../src/plots/FunnelChart';
import { render, cleanup } from '@testing-library/react';

const MOCK_DATA = [
  { action: '浏览网站', pv: 50000, quarter: '2020Q1' },
  { action: '放入购物车', pv: 35000, quarter: '2020Q1' },
  { action: '生成订单', pv: 25000, quarter: '2020Q1' },
  { action: '支付订单', pv: 15000, quarter: '2020Q1' },
  { action: '完成交易', pv: 11500, quarter: '2020Q1' },
  { action: '浏览网站', pv: 80000, quarter: '2020Q2' },
  { action: '放入购物车', pv: 63000, quarter: '2020Q2' },
  { action: '生成订单', pv: 47000, quarter: '2020Q2' },
  { action: '支付订单', pv: 24000, quarter: '2020Q2' },
  { action: '完成交易', pv: 17500, quarter: '2020Q2' },
];


describe('Plots-FunnelChart', () => {
  test('旧版漏斗图 transpose --> isTransposed', () => {
    let chart = null;
    render(<FunnelChart
      data={MOCK_DATA}
      xField='action'
      yField='pv'
      compareField='quarter'
      transpose
      onGetG2Instance={
        (c) => chart = c
      }
    />)
    expect(chart.options).toMatchSnapshot();
    expect(chart.options.isTransposed).toBe(true);
    cleanup();
  })
  test('基础漏斗图', () => {
    let chart = null;
    render(<FunnelChart
      data={MOCK_DATA}
      xField='action'
      yField='pv'
      compareField='quarter'
      onGetG2Instance={
        (c) => chart = c
      }
    />);
    expect(chart.options).toMatchSnapshot();
    cleanup();
  });
  test('水平漏斗图', () => {
    let chart = null;
    render(<FunnelChart
      data={MOCK_DATA}
      xField='action'
      yField='pv'
      compareField='quarter'
      isTransposed
      onGetG2Instance={
        (c) => chart = c
      }
    />);
    expect(chart.options).toMatchSnapshot();
    cleanup();
  });
})
