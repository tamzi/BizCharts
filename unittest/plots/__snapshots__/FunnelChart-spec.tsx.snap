// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plots-FunnelChart 基础漏斗图 1`] = `
Object {
  "appendPadding": Array [
    10,
    5,
    10,
    10,
  ],
  "autoFit": true,
  "compareField": "quarter",
  "data": Array [
    Object {
      "$$conversion$$": Array [
        undefined,
        50000,
      ],
      "$$mappingValue$$": 0.625,
      "$$percentage$$": 0.625,
      "action": "浏览网站",
      "pv": 50000,
      "quarter": "2020Q1",
    },
    Object {
      "$$conversion$$": Array [
        50000,
        35000,
      ],
      "$$mappingValue$$": 0.4375,
      "$$percentage$$": 0.4375,
      "action": "放入购物车",
      "pv": 35000,
      "quarter": "2020Q1",
    },
    Object {
      "$$conversion$$": Array [
        35000,
        25000,
      ],
      "$$mappingValue$$": 0.3125,
      "$$percentage$$": 0.3125,
      "action": "生成订单",
      "pv": 25000,
      "quarter": "2020Q1",
    },
    Object {
      "$$conversion$$": Array [
        25000,
        15000,
      ],
      "$$mappingValue$$": 0.1875,
      "$$percentage$$": 0.1875,
      "action": "支付订单",
      "pv": 15000,
      "quarter": "2020Q1",
    },
    Object {
      "$$conversion$$": Array [
        15000,
        11500,
      ],
      "$$mappingValue$$": 0.14375,
      "$$percentage$$": 0.14375,
      "action": "完成交易",
      "pv": 11500,
      "quarter": "2020Q1",
    },
    Object {
      "$$conversion$$": Array [
        undefined,
        80000,
      ],
      "$$mappingValue$$": 1,
      "$$percentage$$": 1,
      "action": "浏览网站",
      "pv": 80000,
      "quarter": "2020Q2",
    },
    Object {
      "$$conversion$$": Array [
        80000,
        63000,
      ],
      "$$mappingValue$$": 0.7875,
      "$$percentage$$": 0.7875,
      "action": "放入购物车",
      "pv": 63000,
      "quarter": "2020Q2",
    },
    Object {
      "$$conversion$$": Array [
        63000,
        47000,
      ],
      "$$mappingValue$$": 0.5875,
      "$$percentage$$": 0.5875,
      "action": "生成订单",
      "pv": 47000,
      "quarter": "2020Q2",
    },
    Object {
      "$$conversion$$": Array [
        47000,
        24000,
      ],
      "$$mappingValue$$": 0.3,
      "$$percentage$$": 0.3,
      "action": "支付订单",
      "pv": 24000,
      "quarter": "2020Q2",
    },
    Object {
      "$$conversion$$": Array [
        24000,
        17500,
      ],
      "$$mappingValue$$": 0.21875,
      "$$percentage$$": 0.21875,
      "action": "完成交易",
      "pv": 17500,
      "quarter": "2020Q2",
    },
  ],
  "xField": "action",
  "yField": "pv",
}
`;

exports[`Plots-FunnelChart 旧版漏斗图 transpose --> isTransposed 1`] = `
Object {
  "appendPadding": Array [
    10,
    5,
    10,
    10,
  ],
  "autoFit": true,
  "compareField": "quarter",
  "data": Array [
    Object {
      "$$conversion$$": Array [
        undefined,
        50000,
      ],
      "$$mappingValue$$": 0.625,
      "$$percentage$$": 0.625,
      "action": "浏览网站",
      "pv": 50000,
      "quarter": "2020Q1",
    },
    Object {
      "$$conversion$$": Array [
        50000,
        35000,
      ],
      "$$mappingValue$$": 0.4375,
      "$$percentage$$": 0.4375,
      "action": "放入购物车",
      "pv": 35000,
      "quarter": "2020Q1",
    },
    Object {
      "$$conversion$$": Array [
        35000,
        25000,
      ],
      "$$mappingValue$$": 0.3125,
      "$$percentage$$": 0.3125,
      "action": "生成订单",
      "pv": 25000,
      "quarter": "2020Q1",
    },
    Object {
      "$$conversion$$": Array [
        25000,
        15000,
      ],
      "$$mappingValue$$": 0.1875,
      "$$percentage$$": 0.1875,
      "action": "支付订单",
      "pv": 15000,
      "quarter": "2020Q1",
    },
    Object {
      "$$conversion$$": Array [
        15000,
        11500,
      ],
      "$$mappingValue$$": 0.14375,
      "$$percentage$$": 0.14375,
      "action": "完成交易",
      "pv": 11500,
      "quarter": "2020Q1",
    },
    Object {
      "$$conversion$$": Array [
        undefined,
        80000,
      ],
      "$$mappingValue$$": 1,
      "$$percentage$$": 1,
      "action": "浏览网站",
      "pv": 80000,
      "quarter": "2020Q2",
    },
    Object {
      "$$conversion$$": Array [
        80000,
        63000,
      ],
      "$$mappingValue$$": 0.7875,
      "$$percentage$$": 0.7875,
      "action": "放入购物车",
      "pv": 63000,
      "quarter": "2020Q2",
    },
    Object {
      "$$conversion$$": Array [
        63000,
        47000,
      ],
      "$$mappingValue$$": 0.5875,
      "$$percentage$$": 0.5875,
      "action": "生成订单",
      "pv": 47000,
      "quarter": "2020Q2",
    },
    Object {
      "$$conversion$$": Array [
        47000,
        24000,
      ],
      "$$mappingValue$$": 0.3,
      "$$percentage$$": 0.3,
      "action": "支付订单",
      "pv": 24000,
      "quarter": "2020Q2",
    },
    Object {
      "$$conversion$$": Array [
        24000,
        17500,
      ],
      "$$mappingValue$$": 0.21875,
      "$$percentage$$": 0.21875,
      "action": "完成交易",
      "pv": 17500,
      "quarter": "2020Q2",
    },
  ],
  "isTransposed": true,
  "transpose": true,
  "xField": "action",
  "yField": "pv",
}
`;

exports[`Plots-FunnelChart 水平漏斗图 1`] = `
Object {
  "appendPadding": Array [
    10,
    5,
    10,
    10,
  ],
  "autoFit": true,
  "compareField": "quarter",
  "data": Array [
    Object {
      "$$conversion$$": Array [
        undefined,
        50000,
      ],
      "$$mappingValue$$": 0.625,
      "$$percentage$$": 0.625,
      "action": "浏览网站",
      "pv": 50000,
      "quarter": "2020Q1",
    },
    Object {
      "$$conversion$$": Array [
        50000,
        35000,
      ],
      "$$mappingValue$$": 0.4375,
      "$$percentage$$": 0.4375,
      "action": "放入购物车",
      "pv": 35000,
      "quarter": "2020Q1",
    },
    Object {
      "$$conversion$$": Array [
        35000,
        25000,
      ],
      "$$mappingValue$$": 0.3125,
      "$$percentage$$": 0.3125,
      "action": "生成订单",
      "pv": 25000,
      "quarter": "2020Q1",
    },
    Object {
      "$$conversion$$": Array [
        25000,
        15000,
      ],
      "$$mappingValue$$": 0.1875,
      "$$percentage$$": 0.1875,
      "action": "支付订单",
      "pv": 15000,
      "quarter": "2020Q1",
    },
    Object {
      "$$conversion$$": Array [
        15000,
        11500,
      ],
      "$$mappingValue$$": 0.14375,
      "$$percentage$$": 0.14375,
      "action": "完成交易",
      "pv": 11500,
      "quarter": "2020Q1",
    },
    Object {
      "$$conversion$$": Array [
        undefined,
        80000,
      ],
      "$$mappingValue$$": 1,
      "$$percentage$$": 1,
      "action": "浏览网站",
      "pv": 80000,
      "quarter": "2020Q2",
    },
    Object {
      "$$conversion$$": Array [
        80000,
        63000,
      ],
      "$$mappingValue$$": 0.7875,
      "$$percentage$$": 0.7875,
      "action": "放入购物车",
      "pv": 63000,
      "quarter": "2020Q2",
    },
    Object {
      "$$conversion$$": Array [
        63000,
        47000,
      ],
      "$$mappingValue$$": 0.5875,
      "$$percentage$$": 0.5875,
      "action": "生成订单",
      "pv": 47000,
      "quarter": "2020Q2",
    },
    Object {
      "$$conversion$$": Array [
        47000,
        24000,
      ],
      "$$mappingValue$$": 0.3,
      "$$percentage$$": 0.3,
      "action": "支付订单",
      "pv": 24000,
      "quarter": "2020Q2",
    },
    Object {
      "$$conversion$$": Array [
        24000,
        17500,
      ],
      "$$mappingValue$$": 0.21875,
      "$$percentage$$": 0.21875,
      "action": "完成交易",
      "pv": 17500,
      "quarter": "2020Q2",
    },
  ],
  "isTransposed": true,
  "xField": "action",
  "yField": "pv",
}
`;
